### ISSUE TITLE
Certain VCS urls throw stacktrace

### ISSUE THREAD

**Issue by c17r** [kennethreitz/pipenv#805]
```shell
$ pipenv install 'git://github.com/alsoicode/django-medusa.git#egg=django-medusa'
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project.
Installing git://github.com/alsoicode/django-medusa.git#egg=django-medusa…
⠇Warning: You installed a VCS dependency in non–editable mode. This will work fine, but sub-depdendencies will not be resolved by $ pipenv lock.
  To enable this sub–dependency functionality, specify that this dependency is editable.                                                                              Collecting django-medusa from git://github.com/alsoicode/django-medusa.git#egg=django-medusa

Error:  An error occurred while installing git://github.com/alsoicode/django-medusa.git#egg=django-medusa!
Exception:
Traceback (most recent call last):
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/commands/install.py", line 335, in run
    wb.build(autobuilding=True)
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/wheel.py", line 749, in build
    self.requirement_set.prepare_files(self.finder)
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/req/req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/req/req_set.py", line 620, in _prepare_file
    session=self.session, hashes=hashes)
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/download.py", line 805, in unpack_url
    unpack_vcs_link(link, location)
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/download.py", line 479, in unpack_vcs_link
    vcs_backend.unpack(location)
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/vcs/__init__.py", line 286, in unpack
    self.obtain(location)
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/vcs/git.py", line 131, in obtain
    url, rev = self.get_url_rev()
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/vcs/git.py", line 272, in get_url_rev
    url, rev = super(Git, self).get_url_rev()
  File "/Users/christian/.virtualenvs/hnhiring-HX-OrkMQ/lib/python3.6/site-packages/pip/vcs/__init__.py", line 138, in get_url_rev
    assert '+' in self.url, error_message % self.url
AssertionError: Sorry, 'git://github.com/alsoicode/django-medusa.git#egg=django-medusa' is a malformed VCS url. The format is <vcs>+<protocol>://<url>, e.g. svn+http://myrepo/svn/MyApp#egg=MyApp
```

changing to `git+http://github.com/alsoicode/django-medusa.git#egg=django-medusa` works but if pipenv is following the same rules as pip then the original one should have worked as well (https://pip.pypa.io/en/stable/reference/pip_install/#git)

**Comment by techalchemy** [comment]
This is a duplicate of the more descriptive #806 and can probably be closed

**Comment by techalchemy** [comment]
Oh I see now that it is a variant.  It may be fixed by #807 but I will have to check in the morning
